mdata/store_mock.go

```diff
@@ -0,0 +1,48 @@
+package mdata
+
+import "github.com/raintank/metrictank/mdata/chunk"
+
+type mockSearchResult struct {
+	chunks []chunk.IterGen
+	err    error
+}
+
+type mockStore struct {
+	CurrCall int
+	Results  []mockSearchResult
+}
+
+func NewMockStore() *mockStore {
+	d := &mockStore{
+		CurrCall: 0,
+		Results:  make([]mockSearchResult, 0),
+	}
+	return d
+}
+
+func (c *mockStore) AddMockResult(chunks []chunk.IterGen, err error) {
+	// copy chunks because we don't want to modify the source
```

Comment:
In this context the source is `chunks`, because that's what gets copied. 
If we return `chunks` without copying it, then we accept the risk that the underlying values get modified by whatever retrieves them via `Search()`